src/peft/utils/save_and_load.py

```diff
@@ -572,4 +579,27 @@ def get_hub_filename(use_safetensors=True):
     else:
         adapters_weights = torch_load(filename, map_location=torch.device(device))
 
-    return adapters_weights
+    if not key_mapping:
+        remapped_adapters_weights = adapters_weights
+    else:
+        # See discussion in https://github.com/huggingface/transformers/pull/38627
+        remapped_adapters_weights = {}
+        prefix = "base_model.model."
+        for key, val in adapters_weights.items():
+            if not key.startswith(prefix):
+                raise ValueError(
+                    "An error occurred while trying to load a PEFT state_dict with key_mapping. This should not "
+                    "happen. Please open an issue on https://github.com/huggingface/peft/issues and report the error."
+                )
```

Comment:
This would fail for VBLoRA, I think, since that implementation stores weights in `base_model.vblora_vector_bank`. Also not sure about adapters that target encoder/decoder models like T5 where there is no `base_model.model` but `base_model.{shared,encoder,decoder}`.

This remembers me of the auxiliary training wrapper saving mechanism. What we're doing there is to iterate over all the possible prefixes of the model (via `model.named_modules()`) and use those as prefix. Would that work here as well?